instrumentation/log4j/log4j-2.13.2/library/build.gradle.kts

```diff
@@ -3,11 +3,15 @@ plugins {
 }
 
 dependencies {
+  api("io.opentelemetry:opentelemetry-sdk-logs:1.9.0-alpha")
```

Comment:
That will make all clients of this library instrumentation depend on SDK. I assume all users must depend only on API.

Reply:
Users need to call `OpenTelemetryLog4j.initialize(SdkLogEmitterProvider)` to wire opentelemetry into the appender so the log SDK is actually part of this module's API.

Reply:
But this reminds me that I need to update the readme to show usage!

Reply:
@iNikem Yeah - we have https://github.com/open-telemetry/opentelemetry-java/issues/3807 to track fixing this.

Reply:
Sleeping on this, I realized we can keep this code here by making this `compileOnly` - anyone adding the appender has to add the SDK to their classpath themselves.

Reply:
Though I still think it might be simpler to continue to prototype with the code in the SDK repo. What do you think?

Reply:
I'm definitely wary about adding an API API to support this. I think `compileOnly` and a reflective check for the presence of an SDK is a better way to go until we have a spec for a new logging API.

Reply:
`compileOnly` dependency on the SDK sounds reasonable to me. 

> Though I still think it might be simpler to continue to prototype with the code in the SDK repo. 

Prototyping in the SDK has the advantage that the SDK and the appender can develop in lockstep, rather than the appender waiting on SDK releases, which is nice. But I think a lot of the SDK flux has been flushed out (of course assuming no significant changes to the experimental log spec) and so I don't expect to see much value from having the appender in the SDK.